### ISSUE TITLE
Containers are not spread across hosts by default

### ISSUE THREAD

**Issue by janeczku** [rancher/rancher#8179]
**Rancher Versions:**
Server: 1.5.1
Scheduler: v0.7.4

**Docker Version:**
1.12.6

**Environment Type: (Cattle/Kubernetes/Swarm/Mesos)**
Cattle

This is possibly a regression from 1.2 https://github.com/rancher/rancher/issues/6412

**Steps to Reproduce:**

Prerequisite: Create an environment with 2 hosts

1. Create a stack
2. Create a services with a scale of 1 an no resource reservations.
3. Wait until service becomes active
4. Repeat steps 2 and 3 until you have 6 services

**Expected result:** 

Each host has 3 containers running

**Actual results:** 

Containers are not evenly spread. One host has double the amount of containers as the other.

![bildschirmfoto 2017-03-15 um 17 30 54](https://cloud.githubusercontent.com/assets/3813921/23959364/4abfb404-09a5-11e7-9c39-e955cc1f1299.png)

![bildschirmfoto 2017-03-15 um 17 31 24](https://cloud.githubusercontent.com/assets/3813921/23959369/4ea000b0-09a5-11e7-9219-9f4be9035382.png)

```
15.3.2017 17:54:17time="2017-03-15T16:54:17Z" level=info msg="New pool amounts on host 99c589fd-0d4e-4ea6-b94d-9a430624d6cf: instanceReservation total: 1000000 used: 8."
15.3.2017 17:54:17time="2017-03-15T16:54:17Z" level=info msg="Reply: Name: reply.1127954343822485318, PreviousIds: [e8c3ad73-6b91-4bac-8d56-a3a334928206], ResourceId: 1iir1416, Data: map[]."
15.3.2017 17:57:17time="2017-03-15T16:57:17Z" level=info msg="Received event: Name: scheduler.release, Event Id: 58be2e19-ad41-4e71-94b4-a50bd6449522, ResourceId : 1iir1414"
15.3.2017 17:57:17time="2017-03-15T16:57:17Z" level=info msg="Releasing [{Resource:instanceReservation Amount:1}] for b8cb69b7-7d8a-497e-afdf-8fdd3e1f9235"
15.3.2017 17:57:17time="2017-03-15T16:57:17Z" level=info msg="New pool amounts on host b8cb69b7-7d8a-497e-afdf-8fdd3e1f9235: instanceReservation total: 1000000 used: 8."
15.3.2017 17:57:17time="2017-03-15T16:57:17Z" level=info msg="Reply: Name: reply.4466274071334766650, PreviousIds: [58be2e19-ad41-4e71-94b4-a50bd6449522], ResourceId: 1iir1414, Data: map[]."
15.3.2017 17:57:17time="2017-03-15T16:57:17Z" level=info msg="Received event: Name: scheduler.release, Event Id: 85e047d5-49ea-4447-9c08-00ed66ace390, ResourceId : 1iir1412"
15.3.2017 17:57:17time="2017-03-15T16:57:17Z" level=info msg="Releasing [{Resource:instanceReservation Amount:1}] for 99c589fd-0d4e-4ea6-b94d-9a430624d6cf"
15.3.2017 17:57:17time="2017-03-15T16:57:17Z" level=info msg="New pool amounts on host 99c589fd-0d4e-4ea6-b94d-9a430624d6cf: instanceReservation total: 1000000 used: 7."
15.3.2017 17:57:17time="2017-03-15T16:57:17Z" level=info msg="Reply: Name: reply.2909960192217134784, PreviousIds: [85e047d5-49ea-4447-9c08-00ed66ace390], ResourceId: 1iir1412, Data: map[]."
15.3.2017 22:18:40time="2017-03-15T21:18:40Z" level=info msg="Received event: Name: scheduler.prioritize, Event Id: cc1a87d8-6922-4339-b825-9c6d8b354c6e, ResourceId : 1i1423"
15.3.2017 22:18:40time="2017-03-15T21:18:40Z" level=info msg="Reply: Name: reply.280837618772106349, PreviousIds: [cc1a87d8-6922-4339-b825-9c6d8b354c6e], ResourceId: 1i1423, Data: map[prioritizedCandidates:[99c589fd-0d4e-4ea6-b94d-9a430624d6cf b8cb69b7-7d8a-497e-afdf-8fdd3e1f9235 0707e7f0-26f1-496c-a976-a695e1330073]]."
15.3.2017 22:18:40time="2017-03-15T21:18:40Z" level=info msg="Received event: Name: scheduler.reserve, Event Id: ee90d7b3-28fd-4313-ac66-49a72c3bc20b, ResourceId : 1i1423"
15.3.2017 22:18:40time="2017-03-15T21:18:40Z" level=info msg="Reserving [{Resource:instanceReservation Amount:1}] for 99c589fd-0d4e-4ea6-b94d-9a430624d6cf. Force=false"
15.3.2017 22:18:40time="2017-03-15T21:18:40Z" level=info msg="New pool amount on host 99c589fd-0d4e-4ea6-b94d-9a430624d6cf: instanceReservation total: 1000000 used: 9."
15.3.2017 22:18:40time="2017-03-15T21:18:40Z" level=info msg="Reply: Name: reply.280837618772106349, PreviousIds: [ee90d7b3-28fd-4313-ac66-49a72c3bc20b], ResourceId: 1i1423, Data: map[]."
15.3.2017 22:18:41time="2017-03-15T21:18:41Z" level=info msg="Received event: Name: scheduler.reserve, Event Id: e5e4df62-1b76-46a9-8c92-0152e2d4faf8, ResourceId : 1i1423"
15.3.2017 22:18:41time="2017-03-15T21:18:41Z" level=info msg="Reserving [] for 99c589fd-0d4e-4ea6-b94d-9a430624d6cf. Force=false"
15.3.2017 22:18:41time="2017-03-15T21:18:41Z" level=info msg="Reply: Name: reply.280837618772106349, PreviousIds: [e5e4df62-1b76-46a9-8c92-0152e2d4faf8], ResourceId: 1i1423, Data: map[]."
15.3.2017 22:18:57time="2017-03-15T21:18:57Z" level=info msg="Received event: Name: scheduler.prioritize, Event Id: 83713f4a-c193-42f3-9090-7448490fbdd4, ResourceId : 1i1424"
15.3.2017 22:18:57time="2017-03-15T21:18:57Z" level=info msg="Reply: Name: reply.5791067470101751498, PreviousIds: [83713f4a-c193-42f3-9090-7448490fbdd4], ResourceId: 1i1424, Data: map[prioritizedCandidates:[0707e7f0-26f1-496c-a976-a695e1330073 99c589fd-0d4e-4ea6-b94d-9a430624d6cf b8cb69b7-7d8a-497e-afdf-8fdd3e1f9235]]."
15.3.2017 22:18:57time="2017-03-15T21:18:57Z" level=info msg="Received event: Name: scheduler.reserve, Event Id: 938c4247-f0f1-4d52-9203-856c6709942b, ResourceId : 1i1424"
15.3.2017 22:18:57time="2017-03-15T21:18:57Z" level=info msg="Reserving [{Resource:instanceReservation Amount:1}] for 0707e7f0-26f1-496c-a976-a695e1330073. Force=false"
15.3.2017 22:18:57time="2017-03-15T21:18:57Z" level=info msg="New pool amount on host 0707e7f0-26f1-496c-a976-a695e1330073: instanceReservation total: 1000000 used: 10."
15.3.2017 22:18:57time="2017-03-15T21:18:57Z" level=info msg="Reply: Name: reply.5791067470101751498, PreviousIds: [938c4247-f0f1-4d52-9203-856c6709942b], ResourceId: 1i1424, Data: map[]."
15.3.2017 22:18:58time="2017-03-15T21:18:58Z" level=info msg="Received event: Name: scheduler.reserve, Event Id: f3b8ff11-2271-447b-877e-55afed6d2f43, ResourceId : 1i1424"
15.3.2017 22:18:58time="2017-03-15T21:18:58Z" level=info msg="Reserving [] for 0707e7f0-26f1-496c-a976-a695e1330073. Force=false"
15.3.2017 22:18:58time="2017-03-15T21:18:58Z" level=info msg="Reply: Name: reply.5791067470101751498, PreviousIds: [f3b8ff11-2271-447b-877e-55afed6d2f43], ResourceId: 1i1424, Data: map[]."
15.3.2017 22:19:12time="2017-03-15T21:19:12Z" level=info msg="Received event: Name: scheduler.prioritize, Event Id: 1be6e79d-a508-4c5e-ac8b-f4de521937dc, ResourceId : 1i1425"
15.3.2017 22:19:12time="2017-03-15T21:19:12Z" level=info msg="Reply: Name: reply.8004450063763436735, PreviousIds: [1be6e79d-a508-4c5e-ac8b-f4de521937dc], ResourceId: 1i1425, Data: map[prioritizedCandidates:[99c589fd-0d4e-4ea6-b94d-9a430624d6cf b8cb69b7-7d8a-497e-afdf-8fdd3e1f9235 0707e7f0-26f1-496c-a976-a695e1330073]]."
15.3.2017 22:19:12time="2017-03-15T21:19:12Z" level=info msg="Received event: Name: scheduler.reserve, Event Id: 207f91ed-d7ac-4cab-ada7-5fcf16c8cb95, ResourceId : 1i1425"
15.3.2017 22:19:12time="2017-03-15T21:19:12Z" level=info msg="Reserving [{Resource:instanceReservation Amount:1}] for 99c589fd-0d4e-4ea6-b94d-9a430624d6cf. Force=false"
15.3.2017 22:19:12time="2017-03-15T21:19:12Z" level=info msg="New pool amount on host 99c589fd-0d4e-4ea6-b94d-9a430624d6cf: instanceReservation total: 1000000 used: 10."
15.3.2017 22:19:12time="2017-03-15T21:19:12Z" level=info msg="Reply: Name: reply.8004450063763436735, PreviousIds: [207f91ed-d7ac-4cab-ada7-5fcf16c8cb95], ResourceId: 1i1425, Data: map[]."
15.3.2017 22:19:13time="2017-03-15T21:19:13Z" level=info msg="Received event: Name: scheduler.reserve, Event Id: 4c689692-1d05-49cc-b557-b7e3eabc6afc, ResourceId : 1i1425"
15.3.2017 22:19:13time="2017-03-15T21:19:13Z" level=info msg="Reserving [] for 99c589fd-0d4e-4ea6-b94d-9a430624d6cf. Force=false"
15.3.2017 22:19:13time="2017-03-15T21:19:13Z" level=info msg="Reply: Name: reply.8004450063763436735, PreviousIds: [4c689692-1d05-49cc-b557-b7e3eabc6afc], ResourceId: 1i1425, Data: map[]."
15.3.2017 22:19:26time="2017-03-15T21:19:26Z" level=info msg="Received event: Name: scheduler.prioritize, Event Id: 313bb2d3-c10a-4171-b068-b297b216725d, ResourceId : 1i1426"
15.3.2017 22:19:26time="2017-03-15T21:19:26Z" level=info msg="Reply: Name: reply.8866021217594418776, PreviousIds: [313bb2d3-c10a-4171-b068-b297b216725d], ResourceId: 1i1426, Data: map[prioritizedCandidates:[b8cb69b7-7d8a-497e-afdf-8fdd3e1f9235 0707e7f0-26f1-496c-a976-a695e1330073 99c589fd-0d4e-4ea6-b94d-9a430624d6cf]]."
15.3.2017 22:19:27time="2017-03-15T21:19:27Z" level=info msg="Received event: Name: scheduler.reserve, Event Id: 6923b04c-949b-463a-bf20-7367b4b50d97, ResourceId : 1i1426"
15.3.2017 22:19:27time="2017-03-15T21:19:27Z" level=info msg="Reserving [{Resource:instanceReservation Amount:1}] for b8cb69b7-7d8a-497e-afdf-8fdd3e1f9235. Force=false"
15.3.2017 22:19:27time="2017-03-15T21:19:27Z" level=info msg="New pool amount on host b8cb69b7-7d8a-497e-afdf-8fdd3e1f9235: instanceReservation total: 1000000 used: 10."
15.3.2017 22:19:27time="2017-03-15T21:19:27Z" level=info msg="Reply: Name: reply.3106650557452056991, PreviousIds: [6923b04c-949b-463a-bf20-7367b4b50d97], ResourceId: 1i1426, Data: map[]."
15.3.2017 22:19:27time="2017-03-15T21:19:27Z" level=info msg="Received event: Name: scheduler.reserve, Event Id: 77d01435-8946-42d3-a3c1-13c7d803cabe, ResourceId : 1i1426"
15.3.2017 22:19:27time="2017-03-15T21:19:27Z" level=info msg="Reserving [] for b8cb69b7-7d8a-497e-afdf-8fdd3e1f9235. Force=false"
15.3.2017 22:19:27time="2017-03-15T21:19:27Z" level=info msg="Reply: Name: reply.4466274071334766650, PreviousIds: [77d01435-8946-42d3-a3c1-13c7d803cabe], ResourceId: 1i1426, Data: map[]."
15.3.2017 22:19:41time="2017-03-15T21:19:41Z" level=info msg="Received event: Name: scheduler.prioritize, Event Id: d04db901-e344-4c46-8508-3ea5325ee957, ResourceId : 1i1427"
15.3.2017 22:19:41time="2017-03-15T21:19:41Z" level=info msg="Reply: Name: reply.8004450063763436735, PreviousIds: [d04db901-e344-4c46-8508-3ea5325ee957], ResourceId: 1i1427, Data: map[prioritizedCandidates:[0707e7f0-26f1-496c-a976-a695e1330073 99c589fd-0d4e-4ea6-b94d-9a430624d6cf b8cb69b7-7d8a-497e-afdf-8fdd3e1f9235]]."
15.3.2017 22:19:41time="2017-03-15T21:19:41Z" level=info msg="Received event: Name: scheduler.reserve, Event Id: 7f83d71c-50e9-4938-9b5c-4c2e09d89fc4, ResourceId : 1i1427"
15.3.2017 22:19:41time="2017-03-15T21:19:41Z" level=info msg="Reserving [{Resource:instanceReservation Amount:1}] for 0707e7f0-26f1-496c-a976-a695e1330073. Force=false"
15.3.2017 22:19:41time="2017-03-15T21:19:41Z" level=info msg="New pool amount on host 0707e7f0-26f1-496c-a976-a695e1330073: instanceReservation total: 1000000 used: 11."
15.3.2017 22:19:41time="2017-03-15T21:19:41Z" level=info msg="Reply: Name: reply.8004450063763436735, PreviousIds: [7f83d71c-50e9-4938-9b5c-4c2e09d89fc4], ResourceId: 1i1427, Data: map[]."
15.3.2017 22:19:42time="2017-03-15T21:19:42Z" level=info msg="Received event: Name: scheduler.reserve, Event Id: 0b48f869-1f06-4486-a4cb-617050e71e4e, ResourceId : 1i1427"
15.3.2017 22:19:42time="2017-03-15T21:19:42Z" level=info msg="Reserving [] for 0707e7f0-26f1-496c-a976-a695e1330073. Force=false"
15.3.2017 22:19:42time="2017-03-15T21:19:42Z" level=info msg="Reply: Name: reply.8004450063763436735, PreviousIds: [0b48f869-1f06-4486-a4cb-617050e71e4e], ResourceId: 1i1427, Data: map[]."
15.3.2017 22:19:55time="2017-03-15T21:19:55Z" level=info msg="Received event: Name: scheduler.prioritize, Event Id: fd02da8f-51cc-4445-924d-3fe2ad85c48d, ResourceId : 1i1428"
15.3.2017 22:19:55time="2017-03-15T21:19:55Z" level=info msg="Reply: Name: reply.6012070662344822759, PreviousIds: [fd02da8f-51cc-4445-924d-3fe2ad85c48d], ResourceId: 1i1428, Data: map[prioritizedCandidates:[99c589fd-0d4e-4ea6-b94d-9a430624d6cf b8cb69b7-7d8a-497e-afdf-8fdd3e1f9235 0707e7f0-26f1-496c-a976-a695e1330073]]."
15.3.2017 22:19:55time="2017-03-15T21:19:55Z" level=info msg="Received event: Name: scheduler.reserve, Event Id: 42dee05a-3c5d-4c03-9a46-339e611553a9, ResourceId : 1i1428"
15.3.2017 22:19:55time="2017-03-15T21:19:55Z" level=info msg="Reserving [{Resource:instanceReservation Amount:1}] for 99c589fd-0d4e-4ea6-b94d-9a430624d6cf. Force=false"
15.3.2017 22:19:55time="2017-03-15T21:19:55Z" level=info msg="New pool amount on host 99c589fd-0d4e-4ea6-b94d-9a430624d6cf: instanceReservation total: 1000000 used: 11."
15.3.2017 22:19:55time="2017-03-15T21:19:55Z" level=info msg="Reply: Name: reply.6012070662344822759, PreviousIds: [42dee05a-3c5d-4c03-9a46-339e611553a9], ResourceId: 1i1428, Data: map[]."
15.3.2017 22:19:56time="2017-03-15T21:19:56Z" level=info msg="Received event: Name: scheduler.reserve, Event Id: 07cbeef0-cb2f-4626-a031-69d5df6e025f, ResourceId : 1i1428"
15.3.2017 22:19:56time="2017-03-15T21:19:56Z" level=info msg="Reserving [] for 99c589fd-0d4e-4ea6-b94d-9a430624d6cf. Force=false"
15.3.2017 22:19:56time="2017-03-15T21:19:56Z" level=info msg="Reply: Name: reply.6012070662344822759, PreviousIds: [07cbeef0-cb2f-4626-a031-69d5df6e025f], ResourceId: 1i1428, Data: map[]."
```

**Comment by janeczku** [comment]
It seems that when the scheduler service is started before all infrastructure services have been deployed on some of the other hosts, the scheduler has a wrong count of containers for these hosts leading to incorrect balancing of containers when deploying new services.

**Comment by deniseschannon** [comment]
Closing in favor of https://github.com/rancher/rancher/issues/8639 as it has details on how to fix.
